tools/src/main/java/pl/bmstefanski/tools/impl/manager/UserManagerImpl.java

```diff
@@ -1,114 +1,99 @@
+    this.server = server;
   }
 
   @Override
-  public User getUser(UUID uuid) {
-    Validate.notNull(uuid);
-
-    User user = uuidUserCache.getIfPresent(uuid);
-
-    if (user == null) {
-      user = uuidUserMap.get(uuid);
+  public Optional<User> getUser(UUID uniqueId) {
+    Validate.notNull(uniqueId, "Player unique id cannot be null!");
 
-      if (user != null) {
-        uuidUserMap.put(uuid, user);
-        return user;
-      }
-      return new UserImpl(uuid);
-    }
-
-    return user;
+    return this.userUniqueIdMap.values()
+      .stream()
+      .filter(user -> user.getUniqueId().equals(uniqueId))
+      .findFirst();
   }
 
   @Override
-  public User getUser(String playerName) {
-    Validate.notNull(playerName);
-
-    User user = nameUserCache.getIfPresent(playerName);
+  public Optional<User> getUser(String playerName) {
+    Validate.notNull(playerName, "Player name cannot be null!");
 
-    if (user == null) {
-      user = nameUserMap.get(playerName);
+    return this.userNameMap.values()
+      .stream()
+      .filter(user -> user.getName().get().equals(playerName))
+      .findFirst();
+  }
 
-      if (user != null) {
-        nameUserMap.put(playerName, user);
-        return user;
-      }
-      return new UserImpl(playerName);
-    }
+  @Override
+  public Optional<User> getUser(Player player) {
+    Validate.notNull(player, "Player cannot be null!");
 
-    return user;
+    return this.getUser(player.getUniqueId());
   }
 
   @Override
   public void addUser(User user) {
     Validate.notNull(user);
 
-    uuidUserCache.put(user.getUUID(), user);
-    uuidUserMap.put(user.getUUID(), user);
-
-    if (user.getName() != null) {
-      nameUserCache.put(user.getName(), user);
-      nameUserMap.put(user.getName(), user);
+    this.userUniqueIdMap.put(user.getUniqueId(), user);
+    if (user.getName().isPresent()) {
+      this.userNameMap.put(user.getName().get(), user);
     }
   }
 
   @Override
-  public void invalidateUser(User user) {
+  public void removeUser(User user) {
     Validate.notNull(user);
 
-    uuidUserCache.invalidate(user.getUUID());
-    nameUserCache.invalidate(user.getName());
-    uuidUserMap.remove(user.getUUID());
-    nameUserMap.remove(user.getName());
+    this.userUniqueIdMap.remove(user.getUniqueId());
+    this.userNameMap.remove(user.getName().get());
   }
 
   @Override
   public Set<User> getOnlinePlayers() {
-    Set<User> users = new HashSet<>();
-    Bukkit.getOnlinePlayers().forEach(user -> users.add(this.getUser(user.getUniqueId())));
-
-    return users;
+    return this.server.getOnlinePlayers().stream()
+      .map(player -> this.getUser(player.getUniqueId()))
+      .filter(Optional::isPresent)
+      .map(Optional::get)
+      .collect(Collectors.toCollection(LinkedHashSet::new));
   }
 
   @Override
-  public void teleportToLocation(User user, Location location) {
+  public void teleportToLocation(User user, Location location) { // todo optional throw
```

Comment:
TODO zazwyczaj się pisze z wielkich liter, tak dla wygodności.

Reply:
Ani to nie byłaby dobra, ani to nie byłaby zła praktyka, to właściwie tylko wielkość liter, nie :wink: 
Ja się do tego stosunkuje tak, że takie TODO jest bardziej widoczne niż todo, to raz.
Dwa, powinno się jakby nie patrzeć pisać ToDo, ale ze względu na te powiększone litery, i na punkt pierwszy - automatycznie sobie "UpperCase'uje" te literki :wink: 

Taki tylko mój POV na pisanie TODOsów, możesz pisać jak chcesz :slightly_smiling_face: